client/my-sites/site-settings/date-time-format/date-time-format-options.jsx

```diff
@@ -0,0 +1,171 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { capitalize, includes } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Card from 'components/card';
+import SectionHeader from 'components/section-header';
+import DateFormatOption from './date-format-option';
+import StartOfWeekOption from './start-of-week-option';
+import TimeFormatOption from './time-format-option';
+import { defaultDateFormats, defaultTimeFormats } from './default-formats';
+import { getLocalizedDate } from './utils';
+import wrapSettingsForm from '../wrap-settings-form';
+
+export class DateTimeFormatOptions extends Component {
+	state = {
+		customDateFormat: false,
+		customTimeFormat: false,
+		isLoadingSettings: true,
+	};
+
+	componentWillReceiveProps( nextProps ) {
+		const {
+			fields: {
+				date_format: dateFormat,
+				time_format: timeFormat,
+			},
+		} = nextProps;
+
+		if ( ! this.state.isLoadingSettings || '' === dateFormat || '' === timeFormat ) {
+			return;
+		}
+
+		this.setState( {
+			customDateFormat: ! includes( defaultDateFormats, dateFormat ),
+			customTimeFormat: ! includes( defaultTimeFormats, timeFormat ),
+			isLoadingSettings: false,
+		} );
+	}
+
+	setFormat = ( name, defaultFormats ) => event => {
+		const { value: format } = event.currentTarget;
+		this.props.updateFields( { [ `${ name }_format` ]: format } );
+		this.setState( {
+			[ `custom${ capitalize( name ) }Format` ]: ! includes( defaultFormats, format ),
+		} );
+	};
+
+	setDateFormat = this.setFormat( 'date', defaultDateFormats );
+
+	setTimeFormat = this.setFormat( 'time', defaultTimeFormats );
+
+	setCustomFormat = name => event => {
+		const { value: format } = event.currentTarget;
+		this.props.updateFields( { [ `${ name }_format` ]: format } );
+		this.setState( {
+			[ `custom${ capitalize( name ) }Format` ]: true,
+		} );
+	};
+
+	setCustomDateFormat = this.setCustomFormat( 'date' );
+
+	setCustomTimeFormat = this.setCustomFormat( 'time' );
+
+	render() {
+		const {
+			fields: {
+				date_format: dateFormat,
+				start_of_week: startOfWeek,
+				time_format: timeFormat,
+				timezone_string: timezoneString,
+			},
+			handleSelect,
+			handleSubmitForm,
+			isRequestingSettings,
+			isSavingSettings,
+			translate,
+		} = this.props;
+
+		const {
+			customDateFormat,
+			customTimeFormat,
+		} = this.state;
+
+		const localizedDate = getLocalizedDate( timezoneString );
+
+		return (
+			<div>
+				<SectionHeader>
+					<Button
+						compact={ true }
+						onClick={ handleSubmitForm }
+						primary={ true }
+						type="submit"
+						disabled={ isRequestingSettings || isSavingSettings }>
+							{ isSavingSettings
```

Comment:
small nitpick: this confused me here (also a big reason I prefer to keep the `>` on the next line) because of the indentation. could we move that closing `>` into alignment with `<Button`?

```js
<Button
	compact={ true }
	…
>
	{ isSavingSettings
		? …
		: …
	}
</Button>
```